plugins/org.eclipse.emf.mwe2.language/src/org/eclipse/emf/mwe2/language/validation/Mwe2Validator.java

```diff
@@ -399,4 +402,30 @@ public boolean isMandatory(JvmIdentifiableElement feature) {
 		return ePackages;
 	}
 
+	@Check
+	public void checkDeprecatedComponent(Component component) {
+		JvmType type = component.getType();
+		if (type instanceof JvmAnnotationTarget && DeprecationUtil.isDeprecated((JvmAnnotationTarget) type)) {
+			warning(
+				"The '" + type.getQualifiedName() + "' is deprecated.",
+				component,
```

Comment:
somehow "everything" gets underlined and not the class name only

Reply:
so should this be REFERRABLE__TYPE

Reply:
Thanks for the feedback @cdietrich ! I updated the PR to use REFERRABLE_TYPE.